1.train_models/log_reg_plate_3_3p_cp_renorm_fs_data/nbconverted/train_model.py

```diff
@@ -0,0 +1,247 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Training a Logistic Regression Model
+
+# ## Imports
+
+# In[ ]:
+
+
+import sys
+import pandas as pd
+import numpy as np
+from pathlib import Path
+from sklearn.preprocessing import LabelEncoder, MinMaxScaler
+from sklearn.linear_model import LogisticRegression
+from sklearn.model_selection import train_test_split
+from joblib import dump
+
+
+# ## Find the git root Directory
+
+# In[ ]:
+
+
+# Get the current working directory
+cwd = Path.cwd()
+
+if (cwd / ".git").is_dir():
+    root_dir = cwd
+
+else:
+    root_dir = None
+    for parent in cwd.parents:
+        if (parent / ".git").is_dir():
+            root_dir = parent
+            break
+
+# Check if a Git root directory was found
+if root_dir is None:
+    raise FileNotFoundError("No Git root directory found.")
+
+
+# ## Import Utils
+
+# In[ ]:
+
+
+sys.path.append(f"{root_dir}/utils")
+import preprocess_utils as ppu
+
+
+# # Seed and Generator for Reproducibility
+
+# In[ ]:
+
+
+rnd_val = 0  # Random value for all seeds
+rng = np.random.default_rng(seed=rnd_val)  # random number generator
+
+
+# # Converting parquet to pandas dataframe
+
+# ## Feature selected plate data
+
+# In[ ]:
+
+
+filename3 = "Plate_3_sc_norm_fs.parquet"
+filename3p = "Plate_3_prime_sc_norm_fs.parquet"
+plate_path = Path(
+    f"{root_dir}/nf1_painting_repo/3.processing_features/data/feature_selected_data"
+)
+
+path3 = plate_path / filename3
+
+path3p = plate_path / filename3p
+
+# Creates an object for accessing the plate 3 normalized data using the path of the data
+po3 = ppu.Preprocess_data(path=path3)
+
+# Creates an object for accessing the plate 3 prime normalized data using the path of the data
+po3p = ppu.Preprocess_data(path=path3p)
+
+plate3df = po3.df  # Returns the dataframe generated by the csv
+plate3pdf = po3p.df  # Returns the dataframe generated by the csv
```

Comment:
consider moving the inline comments above the code

Reply:
Changed!